maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/report/StatelessXmlReporter.java

```diff
@@ -502,7 +502,7 @@ private static String extraEscape( String message, boolean attribute )
 
         private int c2;
 
-        public EncodingOutputStream( OutputStream out )
+        EncodingOutputStream( OutputStream out )
```

Comment:
why removing public constructor this is not backward compat???? if checkstyle complains about this so remove checkstyle usage as it's useless....

Reply:
This constructor wasn't used outside the package. Intellij also confirms it, so checkstyle reports it properly. Also keep in mind that tests passes. If it not being used, access can be safly lowered to package level.